main_test.go

```diff
@@ -142,6 +142,83 @@ func TestStartProcessors(t *testing.T) {
 	}
 }
 
+func TestRabbitReconnect(t *testing.T) {
+
+	// Start up the processors normally
+	processors, err := StartProcessors(ctx, cfg, make(chan error))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	// Take the first processor
+	processor := processors[0]
+
+	// Set up a rabbit channel closure notification channel
+	channelErrChan := make(chan *amqp.Error)
+	processor.RabbitChannel.NotifyClose(channelErrChan)
```

Comment:
we're setting up the NotifyClose here but later on we're trying to test this by publishing an un-routable message, i thought the NotifyReturn channel deals with this? notifyClose implies that a channel is being closed due to an un-routable message which doesn't seem right 🤔

Reply:
I was seeing this behaviour, although I agree it doesn't seem quite right, I was working under the assumption that just killing the channel is the default behaviour that can be overridden by handling with `NofityReturn`. That said it would be nice to test this in a less weird way, as we discussed in a call I'll look at doing it by making an API call to rabbit and deleting the channels instead.